pdns/reczones.cc

```diff
@@ -168,16 +173,14 @@ bool primeHints(time_t ignored)
 // servers are authoritative for root-servers.net, and some
 // implementations reply not with a delegation on a root-servers.net
 // DS query, but with a NODATA response (the domain is unsigned).
-void primeRootNSZones(bool dnssecmode, unsigned int depth)
+void primeRootNSZones(DNSSECMode mode, unsigned int depth)
 {
   struct timeval now;
   gettimeofday(&now, 0);
   SyncRes sr(now);
 
-  if (dnssecmode) {
-    sr.setDoDNSSEC(true);
-    sr.setDNSSECValidationRequested(true);
-  }
+  sr.setDoDNSSEC(mode != DNSSECMode::Off);
+  sr.setDNSSECValidationRequested(mode != DNSSECMode::Off && mode != DNSSECMode::ProcessNoValidate);
```

Comment:
Am I correctly understanding that we used to do DNSSEC validation in `process` mode and will no longer do so? If so that seems to make sense, and at a quick glance I don't think it will cause any issue.

Reply:
yes, I noticed that when working on this. In `process` mode we should default to on-demand validation.

Reply:
Looking at this again, it think it is more subtle: we were doing immediate validation in `ProcessNoValidate` mode. Which is wrong (but not harmful afaiks) . For `Process` it's debatable if we should call `setDNSSECValidationRequested`.

But the issue is also a bit mood, as the is the function that refreshes the domain information of the root servers names (`root-servers.net`) and not the update of the root NS records themselves and `root-servers.net` is a `Insecure` domain anyway.

This function was needed because of #6138. I now wonder if it still is relevant with the new approach to determining zone-cuts.